src/trace/segment.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Base64 } from 'js-base64';
+import xhrInterceptor from '../interceptors/xhr';
+import uuid from '../services/uuid';
+import Report from '../services/report';
+import { SegmentFeilds, SpanFeilds } from './type';
+import { SpanLayer, SpanType } from '../services/constant';
+import { CustomOptionsType } from '../types';
+
+export default function traceSegment(options: CustomOptionsType) {
```

Comment:
You haven't provided documentation yet, but from here, I think the trace segment will include one and only one span related to RPC, right?
___
Do you have any design/idea about how to make users to inject some spans related to this RPC? Such as, could user create a tracingContext(unfinished segment), and inject this here.
The segment only finished when the tracingContext finished(no active span).

We could add this later, but we should consider this as another core feature.

Reply:
> You haven't provided documentation yet, but from here, I think the trace segment will include one and only one span related to RPC, right?
> 
Emm....No, the trace segment include spans of a service.

Reply:
Let's don't share the same traces with different requests of the same page in default. We could keep this as an option. But further more, we really should consider the manual tracingContext link mechanism.

Reply:
> Let's don't share the same traces with different requests of the same page in default. We could keep this as an option. But further more, we really should consider the manual tracingContext link mechanism.

I agree with this. That's why I asked 👇 , in an SPA web app, there may be many many spans in a browser session, 100+ is totally possible, which may cause trouble in skywalking-rocketbot-ui

> After reviewing the codes, one thing I want to confirm is that, all the spans in one "browser session" belong to a trace (same trace id), right?

Reply:
> After reviewing the codes, one thing I want to confirm is that, all the spans in one "browser session" belong to a trace (same trace id), right?

This needs to be fixed :) But this PR is closing to be ready, super cool. We need to consider a new English blog about this.

Reply:
The previous logic has been modified. Please check screenshots https://github.com/apache/skywalking-client-js/pull/23#issue-545250734